diffrax/brownian/base.py

```diff
@@ -1,14 +1,49 @@
 import abc
+from typing import Optional, Union
 
-from ..custom_types import Array, PyTree, Scalar
+import equinox as eqx
+import jax
+from jax import tree_util as jtu
+
+from ..custom_types import Array, LevyVal, PyTree, Scalar
 from ..path import AbstractPath
 
 
+def _levy_tree_transpose(tree_shape, spacetime_levyarea, tree):
+    """Helper that takes a PyTree of LevyVals and transposes
+    into a LevyVal of PyTrees.
+
+    Args:
+        tree_shape:
+        spacetime_levyarea:
+        tree:
+
+    Returns:
+
+    """
+    hh_default_val = 0.0 if spacetime_levyarea else None
+    return jtu.tree_transpose(
+        outer_treedef=jax.tree_structure(tree_shape),
+        inner_treedef=jax.tree_structure(
+            LevyVal(h=0.0, W=0.0, J=None, H=hh_default_val)
+        ),
+        pytree_to_transpose=tree,
+    )
+
+
 class AbstractBrownianPath(AbstractPath):
-    "Abstract base class for all Brownian paths."
+    """Abstract base class for all Brownian paths."""
+
+    spacetime_levyarea: bool = eqx.field(static=True)
```

Comment:
I think it's generally an antipattern to put fields on abstract base classes.

Are you looking to declare that there must be a `spacetime_levyarea` attribute? (E.g. that the following should typecheck:
```python
def f(x: AbstractBrownianPath):
    y = x.spacetime_levyarea
```
)
or something else?

Reply:
Yes, that was exactly the intention. It is not strictly needed except in some assert statements, but I think it would be useful to have. I suppose I can also use hasattr() first.

Reply:
I removed this. In addition, in VBT and UBP I swapped this out for levy_area: str, which can be either "", "space-time", or eventually "space-time-time".